### PR TITLE
Add instances label to prober metrics

### PR DESCRIPTION
**Pull request by zugao**
## Summary

This PR adds instances label to the prober metrics. This will allow to distinguish between HA and non HA instances.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,6 +17,7 @@ require (
 	github.com/golang/mock v1.6.0
 	github.com/golang/protobuf v1.5.3
 	github.com/hashicorp/go-version v1.6.0
+	github.com/huang-wei/kubernetes v1.14.9
```
Comment (Kidswiss) [inline go.mod:20]: Why did remove `k8s.io/kubernetes v1.26.3` and add this one?

Comment (zugao) [reply go.mod:20]: Had some issues with my IDE, as discussed in the chat I will port in our code the necessary code.

Comment (zugao) [reply go.mod]: Ported everything in `apis/custom`

**Review by Kidswiss — APPROVED** [review]

File: apis/custom/custom_types.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package custom
```
[the remark this answers is not in the document]
Comment (zugao) [reply apis/custom/custom_types.go]: update the name, thanks.

**Comment by Kidswiss** [comment]
I just had a though, while looking at https://github.com/vshn/component-appcat/pull/231

Why not just expose a label `HA` which would be a boolean? Because we have multiple values that could indicate HA setups. For PostgreSQL that would be 2 and 3 instances. And that could get very painful to relfect in the prom queries.

**Comment by zugao** [comment]
You made a good point and I decided to change how we export the HA value.

**Review by Kidswiss — APPROVED** [review]
LGTM

With this we've already future proofed this logic a bit.

**Pull request merged**